src/policies/objective.js

```diff
@@ -17,6 +18,18 @@ export default class Objective {
     return !isUndefined(permissions);
   }
 
+  canUpload() {
+    if (this.objective.status !== OBJECTIVE_STATUS.COMPLETE
+          && (this.objective.otherEntityId
+            || (this.objective.goal
+                && this.objective.goal.status !== GOAL_STATUS.CLOSED
+                && this.objective.goal.grant
```

Comment:
Do we need to check if the grant is active?

Reply:
We can... what are your concerns? In general, I wouldn't want them to pick an objective from the dropdown and get another mysterious error message, but if you are worried about that case we can add the check now and touch up the UX as we go.